rust/src/core.rs

```diff
@@ -39,6 +39,35 @@ pub const STREAM_GAP:      u8 = 0x10;
 pub const STREAM_DEPTH:    u8 = 0x20;
 pub const STREAM_MIDSTREAM:u8 = 0x40;
 
+#[repr(C)]
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum Direction {
+    ToServer = 0x04,
+    ToClient = 0x08,
+}
+
+impl Default for Direction {
+    fn default() -> Self { Direction::ToServer }
+}
```

Comment:
I'm a little worried about using a default that has meaning.  Would adding Unknown to the enum as 0 make sense and using that as the default?

Reply:
I guess a lot of our code just checks for one direction, the blindly assumes the else branch is the other direction. Any thoughts here @victorjulien ?

Reply:
There should be no other direction than these 2, so any other setting is a bug. Would hate to have this dynamically checked all the time as this *should* be unreachable and thus untested code paths.